src/17-infinite-discovery/01-user.ts

```diff
@@ -0,0 +1,87 @@
+import weaviate, { generateUuid5 } from "weaviate-ts-client"
+import _ from "lodash"
+import dotenv from "dotenv"
+import { UsersClassName, User } from "./types"
+import { deleteIfExists } from "system/weaviate"
+
+// Config
+dotenv.config()
+
+// Constants
+const CLASS_NAME: UsersClassName = "InfiniteDiscoveryUsers"
+const USERS: User[] = []
+const BATCH_SIZE: number = 10
+
+const client = weaviate.client({
+  host: process.env.WEAVIATE_URL!,
+})
+
+async function main() {
+  await prepareCollection(CLASS_NAME)
+  await insertObjects(USERS, BATCH_SIZE)
+}
+
+main()
+
+async function prepareCollection(className: UsersClassName) {
+  await deleteIfExists(className)
+
+  const classSchema = {
+    class: className,
+    moduleConfig: {
+      "ref2vec-centroid": {
+        referenceProperties: ["likedArtworks"],
+      },
+    },
+    vectorizer: "ref2vec-centroid",
+    properties: [
+      {
+        name: "internalID",
+        dataType: ["string"],
+        description: "Artsy's internal ID for the user.",
+      },
+      {
+        name: "likedArtworks",
+        dataType: ["InfiniteDiscoveryArtworks"],
+        description:
+          "Artworks liked by this user. Used to calculate the user's vector.",
+      },
+      {
+        name: "DislikedArtworks",
```

Comment:
This is one potential option to track the artworks "disliked" by a user during the discovery flow. As discussed, `ref2vec` doesn't have a way of incorporating "negative" signals (as far as I'm aware) and `weaviate` queries could still potentially include artworks that a user has disliked. And because `weaviate`'s filter operator doesn't have a `NOT LIKE` [option](https://weaviate.io/developers/weaviate/api/graphql/filters#filter-structure) (see note at bottom section), we will need to handle these manually ourselves. For example, by querying `weaviate` for the user, getting the list of `disliked` artworks, and then filtering them from `weaviate`'s [`nearObject`](https://weaviate.io/developers/weaviate/api/graphql/search-operators#nearobject) search results.  

Again, this is just A way and highlights the work we would need to do to filter query results. We should consider others.

Reply:
We can store negative signals in the 'disliked collection' and positive signals in the 'liked collection' in Gravity. I’m not entirely sure if this is feasible in Weaviate, but we could leverage the dislikes from the Gravity collection, which will also be used to power other recommendations. Could this be the intermediate solution?

Reply:
cc @artsy/data-platform - flagging this for the negative signals

Reply:
I see... @mc-jones Are you saying we can't combine `ref2vec` with `moveAwayFrom`? I thought we were using the latter for negative signals.

Reply:
Ah, seems like `moveAwayFrom` is an option on the [nearText](https://weaviate.io/developers/weaviate/api/graphql/search-operators#neartext) query, which we're not using here.

Reply:
@cavvia - Exactly. I think the big question here is if positive signals (with seen artworks filtered out) is able to produce strong enough recommendations. We should still be able to track these signals in analytics or elsewhere, if we want, but they won't be part of the `weaviate` recommendations.

Reply:
Ugh, just noticed this has the capital 'D', it shouldn't be the issue causing this error when deleting the reference from dislikedArtworks?

```
{
	"error": [
		{
			"message": "msg:repo.putobject code:500 err:import into index infinitediscoveryusers: put local object: shard=\"duE1dBoIuudB\": store object in LSM store: Validate vector index for daa63f9d-c771-52fe-9a75-12b643d6c0f1: new node has a vector with length 0. Existing nodes have vectors with length 1536"
		}
	]
}
```

Reply:
@egdbear - fixed this in a [follow-up PR](https://github.com/artsy/quantum/commit/649bebd3ebd20d9acd23b14c933114fbe0e15a48)